### PR TITLE
Clearer code examples on docs main page

### PR DESCRIPTION
**Pull request by cgnorthcutt**
Updated the label errors and learning with noisy labels code examples to better reflect the the generality and usability of the package (with clearer example variable names and comments that help the user understand what's going on).

Also made a few minor tweaks to other language.

### CODE REVIEW

File: docs/source/index.rst [hunk 1]
```diff
@@ -41,16 +41,21 @@ Setting `return_indices_ranked_by` in this function instructs cleanlab to return
...
-   ordered_label_issues = find_label_issues(
-       labels=labels,
-       pred_probs=pred_probs,
-       return_indices_ranked_by='self_confidence',
-   )
+    from cleanlab.classification import CleanLearning
+    from cleanlab.filter import find_label_issues
+    
+    # Option 1 - sklearn-compatible models - just provide the data and labels ツ
+    issues_df = CleanLearning(clf=sklearnCompatibleModel).find_label_issues(data, labels)
```
Comment (jwmueller) [inline docs/source/index.rst:48]: what about either: moving this CleanLearning portion into to its own block, or explaining it a bit more in the text above?  Currently the text above is all about `filter.find_label_issues` but then this code block jumps into CleanLearning first before presenting the `filter` code that was discussed in the text above

File: docs/source/index.rst [hunk 2]
```diff
@@ -67,9 +72,11 @@ When the :py:meth:`.fit() <cleanlab.classification.CleanLearning.fit>` method is
    from sklearn.linear_model import LogisticRegression
    from cleanlab.classification import CleanLearning
 
-   clf = LogisticRegression() # any classifier implementing the sklearn API
-   cl = CleanLearning(clf=clf)
-   cl.fit(X=X, labels=labels)
+   # Train an ML model with noisy labels - 3 lines of code
```
Comment (jwmueller) [inline docs/source/index.rst:75]: if you're trying to save space, this comment may not be needed given it's the section title above this

File: docs/source/index.rst (hunk 1 above)
Comment (cgnorthcutt) [reply docs/source/index.rst:48]: I actually think there is 'too much' text above it. I think these examples are strong stand-alone, as in this stackoverflow post: https://stackoverflow.com/questions/40009134/dealing-with-noisy-training-labels-in-text-classification-using-deep-learning/53604373#53604373

Comment (cgnorthcutt) [reply docs/source/index.rst:48]: @jwmueller i can probably get the text above down to 50%, but it will take an hour to do well and want to make sure thats cool w/ you before i spend the time.

Comment (jwmueller) [reply docs/source/index.rst:48]: ya that sounds worthwhile, especially to avoid the confusion that new CleanLearning is being presented first. I agree the text above can be minimal

**Comment by elisno** [comment]
Failed checks unrelated, fixed in #444.

**Review by elisno — COMMENTED** [review]
Rendered docs
image

File: docs/source/index.rst [hunk 3]
```diff
@@ -35,22 +35,30 @@ Quickstart
...
+cleanlab finds issues in *any dataset that a classifier can be trained on*. The cleanlab package *works with any model* by using model outputs (predicted probabilities) as input -- it doesn't depend on which model created those outputs.
 
-Setting `return_indices_ranked_by` in this function instructs cleanlab to return the indices of potential mislabeled examples, ordered by the likelihood of their given label being incorrect. This is estimated via a *label quality score*, which for example can be specified as ``'self_confidence'`` (predicted probability the given label).
+If you're using a scikit-learn-compatible model (option 1), you don't need to train a model -- you can pass the model, data, and labels into :py:meth:`CleanLearning.find_label_issues <cleanlab.classification.CleanLearning.find_label_issues>` and cleanlab will handle model training for you. If you want to use any non-sklearn-compatible model, e.g. from huggingface, keras, pytorch, etc. (option 2), you can input the trained model's out-of-sample predicted probabilities into :py:meth:`find_label_issues <cleanlab.filter.find_label_issues>`. Examples for both options are below.
```
Comment (jwmueller) [inline docs/source/index.rst:40]:
```suggestion
If you're using a scikit-learn-compatible model (option 1), you don't need to train a model -- you can pass the model, data, and labels into :py:meth:`CleanLearning.find_label_issues <cleanlab.classification.CleanLearning.find_label_issues>` and cleanlab will handle model training for you. If you want to use any non-sklearn-compatible model (option 2), you can input the trained model's out-of-sample predicted probabilities into :py:meth:`find_label_issues <cleanlab.filter.find_label_issues>`. Examples for both options are below.
```

Comment (jwmueller) [reply docs/source/index.rst:40]: we do have sklearn-compatible examples of: huggingface, keras, pytorch

**Review by jwmueller — APPROVED** [review]
LGTM just had a minor suggestion

File: docs/source/index.rst (hunk 3 above)
Comment (cgnorthcutt) [reply docs/source/index.rst:40]: reasoning? examples are helpful since lots of people use those package and may be curious if they will work with cleanlab. or do you see an error?

Comment (jwmueller) [reply docs/source/index.rst:40]: because it makes it sound to me like these cannot be used with cleanlearning when we have specific examples that show how to use each of these with cleanlearning

File: docs/source/index.rst [hunk 4]
```diff
@@ -35,22 +35,31 @@ Quickstart
...
-       return_indices_ranked_by='self_confidence',
-   )
+    # Option 2 - works with ANY ML model - just input the model's predicted probabilities
+    ordered_label_issues = find_label_issues(
+        labels=labels,
+        pred_probs=pred_probs,  # out-of-sample predicted probabilities from any model
+        return_indices_ranked_by='self_confidence',
+    )
+
+:py:class:`CleanLearning <cleanlab.classification.CleanLearning>` (Option 1) also works with models from most standard ML frameworks by wrapping the model for scikit-learn compliance, e.g. huggingface (using KerasWrapperModel), Keras (works by default), pytorch (using skorch), etc.
```
Comment (jwmueller) [inline docs/source/index.rst:57]:
```suggestion
:py:class:`CleanLearning <cleanlab.classification.CleanLearning>` (option 1) also works with models from most standard ML frameworks by wrapping the model for scikit-learn compliance, e.g. huggingface/tensorflow/keras (using our KerasWrapperModel), pytorch (using skorch package), etc. 
```

File: docs/source/index.rst [hunk 5]
```diff
@@ -35,22 +35,31 @@ Quickstart
...
+    # Option 2 - works with ANY ML model - just input the model's predicted probabilities
+    ordered_label_issues = find_label_issues(
+        labels=labels,
+        pred_probs=pred_probs,  # out-of-sample predicted probabilities from any model
+        return_indices_ranked_by='self_confidence',
+    )
+
+:py:class:`CleanLearning <cleanlab.classification.CleanLearning>` (Option 1) also works with models from most standard ML frameworks by wrapping the model for scikit-learn compliance, e.g. huggingface (using KerasWrapperModel), Keras (works by default), pytorch (using skorch), etc. 
+
+By default, :py:meth:`find_label_issues <cleanlab.filter.find_label_issues>` returns a boolean mask of label issues. You can instead return the indices of potential mislabeled examples by setting `return_indices_ranked_by` in :py:meth:`find_label_issues <cleanlab.filter.find_label_issues>`. The indices are ordered by likelihood of a label issue (estimated via :py:meth:`rank.get_label_quality_scores <cleanlab.rank.get_label_quality_scores>`).
```
Comment (jwmueller) [inline docs/source/index.rst:59]:
```suggestion
By default, :py:meth:`find_label_issues <cleanlab.filter.find_label_issues>` returns a boolean mask of label issues. You can instead return the indices of potential mislabeled examples by setting `return_indices_ranked_by` in :py:meth:`find_label_issues <cleanlab.filter.find_label_issues>`. The indices are ordered by likelihood of a label error (estimated via :py:meth:`rank.get_label_quality_scores <cleanlab.rank.get_label_quality_scores>`).
```

File: docs/source/index.rst [hunk 6]
```diff
@@ -67,9 +76,11 @@ When the :py:meth:`.fit() <cleanlab.classification.CleanLearning.fit>` method is
    from sklearn.linear_model import LogisticRegression
    from cleanlab.classification import CleanLearning
 
-   clf = LogisticRegression() # any classifier implementing the sklearn API
-   cl = CleanLearning(clf=clf)
-   cl.fit(X=X, labels=labels)
+   cl = CleanLearning(clf=LogisticRegression())  # any sklearn-compatible classifier
+   cl.fit(X=train_data, labels=labels)
```
Comment (jwmueller) [inline docs/source/index.rst:80]:
```suggestion
   cl.fit(train_data, labels)
```

File: docs/source/index.rst [hunk 7]
```diff
@@ -35,22 +35,31 @@ Quickstart
...
+If you're using a scikit-learn-compatible model (option 1), you don't need to train a model -- you can pass the model, data, and labels into :py:meth:`CleanLearning.find_label_issues <cleanlab.classification.CleanLearning.find_label_issues>` and cleanlab will handle model training for you. If you want to use any non-sklearn-compatible model (option 2), you can input the trained model's out-of-sample predicted probabilities into :py:meth:`find_label_issues <cleanlab.filter.find_label_issues>`. Examples for both options are below.
 
 .. code-block:: python
 
-   from cleanlab.filter import find_label_issues
+    from cleanlab.classification import CleanLearning
+    from cleanlab.filter import find_label_issues
+    
+    # Option 1 - works with sklearn-compatible models - just input the data and labels ツ
+    issues_df = CleanLearning(clf=sklearn_compatible_model).find_label_issues(data, labels)
```
Comment (jwmueller) [inline docs/source/index.rst:48]:
```suggestion
    label_issues_info = CleanLearning(clf=sklearn_compatible_model).find_label_issues(data, labels)
```

File: docs/source/index.rst [hunk 8]
```diff
@@ -67,9 +76,11 @@ When the :py:meth:`.fit() <cleanlab.classification.CleanLearning.fit>` method is
    from sklearn.linear_model import LogisticRegression
    from cleanlab.classification import CleanLearning
 
-   clf = LogisticRegression() # any classifier implementing the sklearn API
-   cl = CleanLearning(clf=clf)
-   cl.fit(X=X, labels=labels)
+   cl = CleanLearning(clf=LogisticRegression())  # any sklearn-compatible classifier
+   cl.fit(X=train_data, labels=labels)
+
+   # Estimate the predictions you would have gotten if you trained with error-free labels.
```
Comment (jwmueller) [inline docs/source/index.rst:82]:
```suggestion
   # Estimate the predictions you would have gotten if you trained without mislabeled data.
```

**Review by jwmueller — APPROVED** [review]
Cleaned it up a little more
